<html>

<!--
Originally hosted at http://ismb2000.sdsc.edu/bosc2000/abstract5.html
Recreated using the snapshots at http://archive.org
-->

<title>BOSC 2000</title>
<style type="text/css"><!--
			   h1 { color: #336; font-style: normal; font-weight: bolder; font-family: Arial, Helvetica }
			   h2 { color: #336; font-style: normal; font-weight: bolder; font-family: Arial, Helvetica }
			   h3 { color: #336; font-style: normal; font-weight: bolder; font-family: Arial, Helvetica }-->
  </style>

<body bgcolor="white">

  <table border="0" cellpadding="2" cellspacing="2">
    <tr>
      <td align="left" valign="top"><img src="bosc2.gif" alt="BOSC 2000" width="100" height="100" hspace="10"></td>
      <td align="center" valign="middle">
	<h1>Telegraph: A Free Library For Probabilistic Sequence Analysis</h1>
	<h3><u>Ian Holmes</u>, Guy Slater and Ewan Birney</h3>
	</td>
      </tr>
    <tr>
      <td colspan="2" align="left" valign="top"><a href="mailto:ihh@fruitfly.org">Ian Holmes &lt;ihh@fruitfly.org&gt;</a><a href="mailto:guy@ebi.ac.uk"><br>
	  Guy Slater &lt;guy@ebi.ac.uk&gt;</a><br>
	<a href="mailto:birney@ebi.ac.uk">Ewan Birney &lt;birney@ebi.ac.uk&gt;</a>
	<p>Hidden Markov Models (HMMs) have been used successfully for a wide range of applications in bioinformatics including protein domain classification, signal peptide recognition and gene prediction, among others. New developments such as Fisher kernels suggest that HMMs have plenty more to offer and that many ideas remain unexplored.</p>
	<p>Many computational biologists express interest in trying out new kinds of HMM architecture for novel problems. In practise, this means writing or otherwise leveraging code to handle:</p>
	<ol>
	  <li type="a">arbitrarily structured HMMs
	    <li type="a">mathematical manipulation of likelihood derivatives and calculus
	      <li type="a">training algorithms
		<li type="a">very fast dynamic programming routines for database searching
		  </ol>
	<p>Several free software packages exist that can do some of these things, but the absence of a free package with all these capabilities means that experimenting with HMMs often necessitates writing a lot of specialised code.</p>
	<p>Telegraph picks up where one previous Open Source project -- Dynamite -- left off. Begun as a collaboration between the EnsEMBL group in Cambridge and the Berkeley Drosophila Genome Project, Telegraph's primary goal was to formalise the functionality of Dynamite while adding capabilities essential for machine learning.</p>
	<p>Enrichments of Telegraph over Dynamite include a supportive infrastructure for working with likelihood calculus (including training, priors, sampling, posterior probabilities and Fisher kernels) as well as a modular design with an XML exchange format linking the higher-level Perl object model to a fast C engine.</p>
	<p>The BOSC and ISMB 2000 conferences coincide with Telegraph's &quot;going public&quot;, and we strongly hope that potential developers, users and testers will contact us during or after the meetings. As an incentive, there will be a repeat of Dynamite's now-legendary offer of a bottle of champagne for the best bugfinder at each release.</p>
	<p>As the number of biologists interested in experimenting with HMMs rises, while the computational demands of the latest algorithms exceed the capabilities of any one centre, the potential for a &quot;lingua franca&quot; in which sequence analysis tools can be described and exchanged is high. We very much hope that you will be interested in becoming involved in any capacity, be it as a user or a developer, and that you'll contact one of the following Telegraph co-ordinators:</p>
	<p></td>
      </tr>
    </table>
  <pre>
</pre>
  </body>

</html>
