<html>

<!--
Originally hosted at http://ismb2000.sdsc.edu/bosc2000/abstract3.html
Recreated using the snapshots at http://archive.org
-->

<title>BOSC 2000</title>
<style type="text/css"><!--
			   h1 { color: #336; font-style: normal; font-weight: bolder; font-family: Arial, Helvetica }
			   h2 { color: #336; font-style: normal; font-weight: bolder; font-family: Arial, Helvetica }
			   h3 { color: #336; font-style: normal; font-weight: bolder; font-family: Arial, Helvetica }-->
  </style>

<body bgcolor="white">

  <table border="0" cellpadding="2" cellspacing="2">
    <tr>
      <td align="left" valign="top"><img src="bosc2.gif" alt="BOSC 2000" width="100" height="100" hspace="10"></td>
      <td align="center" valign="middle">
	<h1>An Introduction to Genome Annotation Markup Elements (GAME) XML</h1>
	<h3>Suzanna Lewis</h3>
	<p></td>
      </tr>
    <tr>
      <td colspan="2" align="left" valign="top">
	<table border="0" cellpadding="0" cellspacing="0" width="100%">
	  <tr>
	    <td>Berkeley Drosophila Genome Project</td>
	    <td align="right"><a href="mailto:suzi@fruitfly.berkeley.edu">suzi@fruitfly.berkeley.edu</a></td>
	    </tr>
	  <tr>
	    <td>539 Life Sciences Addition</td>
	    <td align="right">Tel: 510-643-0514</td>
	    </tr>
	  <tr>
	    <td>University of California</td>
	    <td align="right">Fax: 510-643-9947</td>
	    </tr>
	  <tr>
	    <td>Berkeley, CA 94720</td>
	    <td align="right"><a href="/web/20000925205152/http://fruitfly.berkeley.edu/">http://fruitfly.berkeley.edu</a></td>
	    </tr>
	  </table>
	<p>This work is focused on defining the essential XML elements necessary to describe biological sequences (both nucleic and protein), results of analytical work carried out on those sequences (both computational and experimental), and to describe the conclusions that are drawn from these analyses. The resulting GAME XML is being used for large-scale annotation efforts being carried out at FlyBase and for EnsEMBL. GAME XML defines these elements at a very high level of abstraction in order to allow for the greatest flexibility and adaptability to a full range of possibilities. The requirements include the ability to describe all varieties of computational analysis, such as alignments (e.g. BLAST, sim4, clustalW, etc.), predictions (e.g. Genie, genscan, genefinder, etc.), motifs (e.g. pfam, BLOCKS, promoter regions, etc.), experimental results (e.g. mutational events, location of chromosomal breakpoints, etc.), and any other feature that can be reduced to an interval (or set of intervals) along the length of a molecular sequence. GAME makes an absolute distinction between the results, which serve as primary data, and the conclusions that are drawn from the results, which are termed 'annotations'. This precise definition of an annotation in GAME demands that every annotation be justified and supported by the results that come from the analysis of the sequence. We will describe the overall structure and explain the logic behind each design decisions that we made. GAME differs from GFF in both the problem it solves and in the richness of the semantics. In practical applications GAME has proven to be successfully flexible to support complex annotations of entire genomes.</td>
      </tr>
    </table>
  <pre>
</pre>
  </body>

</html>
