<html>

<!--
Originally hosted at http://ismb2000.sdsc.edu/bosc2000/abstract1.html
Recreated using the snapshots at http://archive.org
-->

<title>BOSC 2000</title>
<style type="text/css"><!--
			   h1 { color: #336; font-style: normal; font-weight: bolder; font-family: Arial, Helvetica }
			   h2 { color: #336; font-style: normal; font-weight: bolder; font-family: Arial, Helvetica }
			   h3 { color: #336; font-style: normal; font-weight: bolder; font-family: Arial, Helvetica }-->
  </style>

<body bgcolor="white">

  <table border="0" cellpadding="2" cellspacing="2">
    <tr>
      <td align="left" valign="top"><img src="bosc2.gif" alt="BOSC 2000" width="100" height="100" hspace="10"></td>
      <td align="center" valign="middle">
	<h1>Distributing Bioinformatics Applications with Piper</h1>
	<h3>J:W:Bizzaro*, Gary Van Domselaar, Brad Chapman, Jean-Marc Valin, Jarl van Katwijk, Dominic Letourneau and Deanne Taylor</h3>
	</td>
      </tr>
    <tr>
      <td colspan="2" align="left" valign="top">* Corresponding author:<br>
	Bioinformatics.org: The Open Lab<br>
	c/o Department of Chemistry<br>
	University of Massachusetts Lowell<br>
	Lowell, MA 01854<br>
	<a href="mailto:jeff@bioinformatics.org">jeff@bioinformatics.org</a>
	<p>A typical problem in bioinformatics research is linking together multiple pro- grams to analyze a set of information. A common example of this is building phylogenetic trees from DNA sequences. The sequences are initially aligned us- ing a sequence alignment program, are then analyzed in a phylogeny program to produce trees, and finally the trees are visualized in a viewer. This process can be further complicated by the fact that the programs may have incompatible inputs and outputs, as well as extensive memory and processing requirements. To address these problems the authors have developed Piper, a distributed plat- form that can be used to link bioinformatics programs. Piper is ideally suited to broker bioinformatics analyses in that it can combine highly specific, modular, data repositories and data analysis functions together to provide sophisticated and effcient data processing networks. Piper provides a wrapper around ex- isting bioinformatics programs so that they can be connected and executed in an intuitive manner. In addition, individual programs can be located on re- mote computers so that expensive calculations can be executed on faster, even dedicated, hardware. Piper is designed as a modular system using CORBA con- nectivity protocols as the backbone to link the modules. This design allows a number of different user interfaces to control the core processing engine. In ad- dition, Piper is being developed under the Open Source model, allowing contri- bution and design feedback from individuals in multiple areas of bioinformatics. Piper is especially well suited for automated, high-throughput data analysis like protein fold identification, sequence conditioning such as repeat/vector masking and data format conversion, and customization of batch scripting processes such as building phylogenetic trees from DNA sequences.</td>
      </tr>
    </table>
  <pre>

</pre>
  </body>

</html>
