<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html>
  <head>
    <title>The Apollo Genome Annotation Tool</title>
  </head>

  <body bgcolor=white>
<center>
    <h1>The Apollo Genome Annotation Tool</h1>

<h3>Nomi L. Harris (nomi@fruitfly.org), Suzanna E. Lewis, Mark Gibson, Colin Wiel, John Richter</h3>
<h4>Berkeley Drosophila Genome Project</h4>
<h3>Stephen M.J. Searle,
Michele E. Clamp</h3>
<h4>The Sanger Institute and the European Bioinformatics Institute
</h4>
</center>

Apollo is an Open Source genome annotation viewer and editor. It was
developed as a collaboration between the Berkeley Drosophila Genome
Project (part of the FlyBase consortium) and The Sanger Institute in
Cambridge, UK. Apollo allows researchers to explore genomic
annotations at many levels of detail, and to perform expert annotation
curation, all in a graphical environment. It is being used by the
FlyBase biologists to make the final annotations on the finished
<i>Drosophila melanogaster</i> genome, and will also be the primary vehicle
for sharing these annotations with the community.

<P>

An increasing number of research groups are using Apollo as a starting
point for customizing their own annotation visualization tool. Because
Apollo was developed from the beginning to serve the needs of two
groups working on different organisms (fruitfly and human) with different
types of data, it was specifically designed to be flexible and
extensible. The Generic Model Organism Database (GMOD) project, which
aims to provide a complete ready-to-use toolkit for analyzing whole
genomes, has adopted Apollo as its annotation workbench.

<P>

Apollo can read annotation data in a variety of formats (including GAME
XML, GFF, and GenBank) via CGI, CORBA, and flat files, and the data
adapter interface makes it straightforward to add new formats.  The "look
and feel" of the display is also highly configurable. For example, rows
of data (known as "tiers") can be moved with the mouse; the threshold for
displaying results can be changed; and the colors of each result can be
customized, all from within Apollo.

<P>

Apollo is a powerful annotation curation workbench, simplifying the task of
biologists who are poring over thousands of computational results and
deciding how to summarize all the relevant data into a complete and
accurate description of the genome and, ultimately, the proteome. Gene
transcripts are easily created by dragging computational results and
dropping them in the annotation zone. Other tools within Apollo let the
curators edit the annotations in many ways, including adding canned or
customized comments. Validation of annotations is simplified by edge
matching, start/stop codons, and splice site detection.  The new Synteny
Viewer allows visualization of cross-genome comparisons.

<P>

Apollo is available
at SourceForge: <a
href=http://sourceforge.net/projects/gmod/>http://sourceforge.net/projects/gmod/</a>.
Like all gmod components, it is distributed under the terms of the
Artistic License.

<P><HR><P>

<!-- Created: Fri Apr 26 11:31:43 PDT 2002 -->
<!-- hhmts start -->
Last modified: Tue Apr 30 11:17:46 PDT 2002
<!-- hhmts end -->
  </body>
</html>
